iib/workers/tasks/opm_operations.py

```diff
@@ -92,13 +78,17 @@ def opm_serve(catalog_dir: str) -> Tuple[int, subprocess.Popen]:
     """
     log.info('Serving data from file-based catalog %s', catalog_dir)
 
-    port = _get_free_port_for_grpc()
-    cmd = ['opm', 'serve', catalog_dir, '-p', str(port), '-t', '/dev/null']
-    cwd = os.path.abspath(os.path.join(catalog_dir, os.path.pardir))
-    return (
-        port,
-        _serve_cmd_at_port_defaults(cmd, cwd, port),
-    )
+    for port in _gen_port_for_grpc():
+        try:
+            cmd = ['opm', 'serve', catalog_dir, '-p', str(port), '-t', '/dev/null']
+            cwd = os.path.abspath(os.path.join(catalog_dir, os.path.pardir))
+            return (
+                port,
+                _serve_cmd_at_port_defaults(cmd, cwd, port),
+            )
+        except AddressAlreadyInUse:
+            log.debug('Port %s is already taken. Checking next one...', port)
+            continue
```

Comment:
Do we still need to have retry decorator if we are catching the exception? I think that now it would run 100x100 times before failing.

Reply:
@xDaile thanks, nice catch.

Reply:
please check